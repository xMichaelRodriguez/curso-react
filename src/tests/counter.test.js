import React from "react";
import { shallow } from "enzyme";
import Counter from "../counterApp";

describe("Pruebas en app", () => {
  let wrapper = shallow(<Counter numero={10} />);
  beforeEach(() => {
    wrapper = shallow(<Counter numero={10} />);
  });

  test("Debe de mostrar el componente <Counter/>", () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el valor defectos", () => {
    const wrapper = shallow(<Counter numero={100} />);
    const parrafo = wrapper.find("h2").text().trim();
    /* console.log(parrafo); */
    expect(parrafo).toBe("100");
  });

  test("Debe incrementer con el boton +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("11");
  });

  test("Debe disminuir con el boton: restar", () => {
    wrapper.find("button").at(2).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("9");
  });

  test("Debe Colocar valor defecto del boton reset",()=>{
    const wrapper = shallow(<Counter numero={105} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe('105')
  })
});
