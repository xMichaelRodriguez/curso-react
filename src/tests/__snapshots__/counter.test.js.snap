// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en app Debe de mostrar el componente <Counter/> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
    type="button"
  >
    Clickeame
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Recet
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    resta
  </button>
</Fragment>
`;
